kmp_match2/kmp_match.cpp

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+/*--- kmp����� ���ڿ�� �˻� ---*/
+int kmp_match(const char txt[], const char pat[])
+{
+	int pt = 1;		/* txt Ŀ�� */
+	int pp = 0;		/* pat Ŀ�� */
+	int skip[1024]; /* �ǳʶٱ� ǥ */
+
+	/* ǥ ����� */
+	skip[pt] = 0;
+	while (pat[pt] != '\0') {
+		if (pat[pt] == pat[pp])
+			skip[++pt] = ++pp;
+		else if (pp == 0)
+			skip[++pt] = pp;
+		else
+			pp = skip[pp];
```

Comment:
이 부분을 출력해 보면 pp의 값에 따라 skip[pp]의 값이 뭐가 들어있는지 확인해 볼 수 있지 않을까요?
```suggestion
		else {
			pp = skip[pp];
			printf("pp=%d\n", pp);
			printf("skip[pp]=%d\n", skip[pp]);
		}
```

Reply:
텍스트 : ZABCABEABCABDAE
패턴 : ABCABD
pp=0
skip[0]=-858993460
pp=0
skip[0]=-858993460
pp=1
skip[1]=0
pp=2
skip[2]=0
pp=0
skip[0]=-858993460
pp=0
skip[0]=-858993460

Reply:
```
		else if (pp == 0) //처음부터 틀리면
			skip[++pt] = pp; //한 칸 뒤로 보낸다.
		else //틀리면
			pp = skip[pp]; //패턴의 비교 위치를 skip[pp]만큼 뒤로 돌린다?
```

Reply:
더 생각해보겠습니다......